src/backend/starknet.cairo

```diff
@@ -183,42 +183,31 @@ namespace Internals {
         if (starknet_account_exists == 0) {
             // Deploy account
             Starknet.deploy(self.address.evm);
-
-            let has_code_or_nonce = Account.has_code_or_nonce(self);
-            if (has_code_or_nonce == FALSE) {
-                // Nothing to commit
-                return ();
-            }
-
-            // If SELFDESTRUCT, leave the account empty after deploying it - including
-            // burning any leftover balance.
-            if (self.selfdestruct != 0) {
-                let starknet_address = Account.compute_starknet_address(Constants.BURN_ADDRESS);
-                tempvar burn_address = new model.Address(
-                    starknet=starknet_address, evm=Constants.BURN_ADDRESS
-                );
-                let transfer = model.Transfer(self.address, burn_address, [self.balance]);
-                State.add_transfer(transfer);
-                return ();
-            }
-
-            // Write bytecode
-            IAccount.write_bytecode(starknet_address, self.code_len, self.code);
-            // Set nonce
-            IAccount.set_nonce(starknet_address, self.nonce);
-            // Save storages
-            _save_storage(starknet_address, self.storage_start, self.storage);
-
-            // Save valid jumpdests
-            Internals._save_valid_jumpdests(
-                starknet_address, self.valid_jumpdests_start, self.valid_jumpdests
-            );
-            return ();
+            tempvar syscall_ptr = syscall_ptr;
+            tempvar pedersen_ptr = pedersen_ptr;
+            tempvar range_check_ptr = range_check_ptr;
+        } else {
+            tempvar syscall_ptr = syscall_ptr;
+            tempvar pedersen_ptr = pedersen_ptr;
+            tempvar range_check_ptr = range_check_ptr;
```

Comment:
if we update the signature of `Starknet.deploy` to not return the (useless) starknet address, we can save 3 steps here

```suggestion
        } else {
            tempvar syscall_ptr = syscall_ptr;
            tempvar pedersen_ptr = pedersen_ptr;
            tempvar range_check_ptr = range_check_ptr;
         }
         let syscall_ptr = cast([ap - 3], felt*);
         let pedersen_ptr = ...
```